Failed to create pull request: Error: TF401398: The pull request cannot be activated because the source and/or the target branch no longer exists, or the requested refs are not branches

**Describe the bug**
Dependabot creates a new branch but can't create a pull-request

Creating pull request 'Bump Serilog from 4.0.1 to 4.0.2'...
 - Pushing 1 change(s) to branch 'dependabot/nuget/refs/heads/master/Serilog-4.0.2'...
 - Creating pull request to merge 'dependabot/nuget/refs/heads/master/Serilog-4.0.2' into 'refs/heads/master'...
##[error]Failed to create pull request: Error: TF401398: The pull request cannot be activated because the source and/or the target branch no longer exists, or the requested refs are not branches
Error: TF401398: The pull request cannot be activated because the source and/or the target branch no longer exists, or the requested refs are not branches
    at RestClient.<anonymous> (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.35.923/node_modules/typed-rest-client/RestClient.js:204:31)
    at Generator.next (<anonymous>)
    at fulfilled (/home/vsts/work/_tasks/dependabot_d98b873d-cf18-41eb-8ff5-234f14697896/2.35.923/node_modules/typed-rest-client/RestClient.js:7:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  statusCode: 400,
  result: {
    '$id': '1',
    innerException: null,
    message: 'TF401398: The pull request cannot be activated because the source and/or the target branch no longer exists, or the requested refs are not branches',
    typeName: 'Microsoft.TeamFoundation.Git.Server.GitPullRequestCannotBeActivated, Microsoft.TeamFoundation.Git.Server',
    typeKey: 'GitPullRequestCannotBeActivated',
    errorCode: 0,
    eventId: 3000
  },
  responseHeaders: {
    'cache-control': 'no-cache',
    pragma: 'no-cache',
    'content-length': '384',
    'content-type': 'application/json; charset=utf-8',
    expires: '-1',
    p3p: 'CP="CAO DSP COR ADMa DEV CONo TELo CUR PSA PSD TAI IVDo OUR SAMi BUS DEM NAV STA UNI COM INT PHY ONL FIN PUR LOC CNT"',
    'x-tfs-processid': 'afad4bcc-8099-4abb-915d-1f9d331d773b',
    'strict-transport-security': 'max-age=31536000; includeSubDomains',
    activityid: 'baf08b20-0b16-481a-a693-192b14133778',
    'x-tfs-session': 'baf08b20-0b16-481a-a693-192b14133778',
    'x-vss-e2eid': 'baf08b20-0b16-481a-a693-192b14133778',
    'x-vss-senderdeploymentid': 'a6c8fbe9-7425-4392-883d-c602e2e7f7eb',
    'x-vss-userdata': '085815d7-e6fe-48de-b063-e7286aacc314:0bab63e6-4a55-40e1-afe0-d2fee468b8a3',
    'x-frame-options': 'SAMEORIGIN',
    'request-context': 'appId=cid-v1:e6292eea-fb85-4107-bc0a-339fd28d3647',
    'access-control-expose-headers': 'Request-Context',
    'x-content-type-options': 'nosniff',
    'x-cache': 'CONFIG_NOCACHE',
    'x-msedge-ref': 'Ref A: 7363C1C86BF14A64921B9012A2FDA414 Ref B: AMS04EDGE2217 Ref C: 2024-09-29T08:10:29Z',
    date: 'Sun, 29 Sep 2024 08:10:29 GMT'
  }
}
Processing output 'mark_as_processed' with data: { 'base-commit-sha': 'd1e623fc2f35d7248292db82c01f992d89c1bae3' }
##[error]1 update job(s) failed, check logs for more information
Finishing: dependabot

--- a/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.ts
+++ b/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.ts
@@ -65,7 +65,8 @@ export class AzureDevOpsWebApiClient {
         throw new Error(`Repository '${project}/${repository}' not found`);
       }
 
-      return repo.defaultBranch;
+      // Strip reference prefix from the branch name, the caller doesn't need to know this
+      return repo.defaultBranch?.toLowerCase()?.replace('refs/heads/', '');
     } catch (e) {
       error(`Failed to get default branch for '${project}/${repository}': ${e}`);
       console.debug(e); // Dump the error stack trace to help with debugging

--- a/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.ts
+++ b/extension/tasks/dependabotV2/utils/azure-devops/AzureDevOpsWebApiClient.ts
@@ -66,7 +66,7 @@ export class AzureDevOpsWebApiClient {
       }
 
       // Strip reference prefix from the branch name, the caller doesn't need to know this
-      return repo.defaultBranch?.toLowerCase()?.replace('refs/heads/', '');
+      return repo.defaultBranch?.replace(/^refs\/heads\//i, '');
     } catch (e) {
       error(`Failed to get default branch for '${project}/${repository}': ${e}`);
       console.debug(e); // Dump the error stack trace to help with debugging

